plugins/user-authenticators/ldap/src/org/apache/cloudstack/ldap/LdapConfiguration.java

```diff
@@ -36,6 +36,8 @@
 
     private static final ConfigKey<Integer> ldapPageSize = new ConfigKey<Integer>(Integer.class, "ldap.request.page.size", "Advanced", "1000",
                                                                                "page size sent to ldap server on each request to get user", true, ConfigKey.Scope.Global, 1);
+    private static final ConfigKey<String> ldapProvider = new ConfigKey<String>(String.class, "ldap.provider", "Advanced", "openldap", "ldap provider ex:openldap, microsoftad",
+                                                                                true, ConfigKey.Scope.Global, null);
```

Comment:
no. multiple providers are not possible. 
I havent touched any of the existing configurations as I will be refactoring them along with https://cwiki.apache.org/confluence/display/CLOUDSTACK/WIP%3A+LDAP%3A+Trust+AD+and+Auto+Import